src/ray/core_worker/task_interface.h

```diff
@@ -0,0 +1,92 @@
+#ifndef RAY_CORE_WORKER_TASK_INTERFACE_H
+#define RAY_CORE_WORKER_TASK_INTERFACE_H
+
+#include "common.h"
+#include "ray/common/io.h"
+#include "ray/id.h"
+#include "ray/status.h"
+
+namespace ray {
+
+class CoreWorker;
+
+/// Options of a task call.
+struct CallOptions {
+  /// Number of returns of this task.
+  int num_returns = 1;
+  /// Resources required by this task.
+  std::unordered_map<std::string, double> resources;
+};
+
+/// Options of an actor creation task.
+struct ActorCreationOptions {
+  /// Maximum number of times that the actor should be reconstructed when it dies
+  /// unexpectedly. It must be non-negative. If it's 0, the actor won't be reconstructed.
+  uint64_t max_reconstructions = 0;
+  /// Resources required by the whole lifetime of this actor.
+  std::unordered_map<std::string, double> resources;
+};
+
+/// A handle to an actor.
+class ActorHandle {
+ public:
+  ActorHandle(const ActorID &actor_id, const ActorHandleID &actor_handle_id)
+      : actor_id_(actor_id), actor_handle_id_(actor_handle_id){};
+
+  /// ID of the actor.
+  const ActorID &ActorID() const { return actor_id_; };
+
+  /// ID of this actor handle.
+  const ActorHandleID &ActorHandleID() const { return actor_handle_id_; };
+
+ private:
+  class ActorID actor_id_;
```

Comment:
why is `class` here? 

also, maybe make these `const`?

we should document all fields before merging